tools/edit_instance_template/edit_instance_template.go

```diff
@@ -51,26 +51,40 @@ func parseKubeLabels(tpl *computepb.InstanceTemplate) map[string]string {
 }
 
 func findInstanceTemplates(ctx context.Context, c *compute.InstanceTemplatesClient) ([]*computepb.InstanceTemplate, error) {
-	it := c.List(ctx, &computepb.ListInstanceTemplatesRequest{
-		Filter:  proto.String(fmt.Sprintf("name = gke-%s-*", *cluster)),
+	filter := fmt.Sprintf("name = gke-%s-*", *cluster)
+	it := c.AggregatedList(ctx, &computepb.AggregatedListInstanceTemplatesRequest{
+		Filter:  proto.String(filter),
 		Project: *project,
 	})
 
+	numTemplates := 0
 	templates := make([]*computepb.InstanceTemplate, 0)
 	for {
-		tpl, err := it.Next()
+		resp, err := it.Next()
 		if err == iterator.Done {
 			break
 		}
 		if err != nil {
 			return nil, err
 		}
-		labels := parseKubeLabels(tpl)
-		if labels["cloud.google.com/gke-nodepool"] != *pool {
-			continue
+		for _, tpl := range resp.Value.InstanceTemplates {
+			numTemplates++
+			labels := parseKubeLabels(tpl)
+			if labels["cloud.google.com/gke-nodepool"] != *pool {
+				log.Debugf("Ignoring instance template %s (wrong nodepool)",
+					tpl.GetName())
+				continue
+			}
+			log.Debugf("Found editable instance template %s", tpl.GetName())
+			templates = append(templates, tpl)
 		}
-		templates = append(templates, tpl)
 	}
+
+	if numTemplates == 0 {
+		log.Fatalf("No instance templates matching '%s' returned by "+
```

Comment:
nit: it's usually bad form to log.Fatalf from outside of main. return an error here, the caller is already calling log.Fatalf on errors.

Reply:
done